Runtime/Triangulator.cs

```diff
@@ -173,187 +234,217 @@ public class InputData
-
-            Input.Positions = tmpInputPositions;
-            tmpInputPositions = default;
-
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                Input.HoleSeeds = tmpInputHoleSeeds;
-                tmpInputHoleSeeds = default;
-            }
-
-            return dependencies;
-        }
-
-        private JobHandle ScheduleWorldToLocalTransformation(JobHandle dependencies)
-        {
-            tmpInputPositions = Input.Positions;
-            Input.Positions = localPositions.AsDeferredJobArray();
-            if (Input.HoleSeeds.IsCreated)
-            {
-                tmpInputHoleSeeds = Input.HoleSeeds;
-                Input.HoleSeeds = localHoleSeeds.AsDeferredJobArray();
-            }
-
-            dependencies = new InitialLocalTransformationJob(this).Schedule(dependencies);
-            if (tmpInputHoleSeeds.IsCreated)
-            {
-                dependencies = new CalculateLocalHoleSeedsJob(this).Schedule(dependencies);
+        [BurstCompile]
+        struct TriangulationJob: IJob {
+            public Preprocessor preprocessor;
+            public bool validateInput;
+            public bool restoreBoundary;
+            public bool refineMesh;
+            public int sloanMaxIters;
+            public float concentricShellsParameter;
+            public float refinementThresholdArea;
+            public float refinementThresholdAngle;
+            public InputData input;
+            public OutputData output;
+
+            public struct InputData {
+                public NativeArray<float2> Positions;
+
+                [NativeDisableContainerSafetyRestriction]
+                public NativeArray<int> ConstraintEdges;
+
+                [NativeDisableContainerSafetyRestriction]
+                public NativeArray<float2> HoleSeeds;
+            }
+
+            // Markers
+            static readonly ProfilerMarker MarkerPreProcess = new ProfilerMarker("PreProcess");
+            static readonly ProfilerMarker MarkerValidateInput = new ProfilerMarker("ValidateInput");
+            static readonly ProfilerMarker MarkerDelaunayTriangulation = new ProfilerMarker("DelaunayTriangulation");
+            static readonly ProfilerMarker MarkerConstrainEdges = new ProfilerMarker("ConstrainEdges");
+            static readonly ProfilerMarker MarkerPlantSeeds = new ProfilerMarker("PlantSeeds");
+            static readonly ProfilerMarker MarkerRefineMesh = new ProfilerMarker("RefineMesh");
+            static readonly ProfilerMarker MarkerInverseTransformation = new ProfilerMarker("InverseTransformation");
+
+            static void PreProcessInput (Preprocessor preprocessor, InputData input, OutputData output, out NativeList<float2> localPositions, out NativeArray<float2> localHoles, out AffineTransform2D localTransformation) {
+                localPositions = output.Positions.IsCreated ? output.Positions : new NativeList<float2>(input.Positions.Length, Allocator.Temp);
```

Comment:
I do not quite get why `output.Positions.IsCreated` can be `false`. By looking only on code it seems that the following list is always allocated.

When I have access to PC, I will debug this.

Reply:
Currently, it is always created. However, in the future I think it would be nice to not have to pass an output array for the positions. This would allow the localPositions array to be completely temporary. This is useful if you are not using any mesh refinement, and thus the input positions will be identical (up to floating point error) to the output positions.

Reply:
Okay, I agree, this can be also performance wise. However I think that it is better to not introduce this now. It is better to introduce this with fully support optional position output.